Compile fails with latest git on macOS

I get a compile error at 
```
[ 33%] Building CXX object CMakeFiles/nomacsCore-3.19.1.dir/src/DkCore/DkUtils.cpp.o
In file included from /Users/hmueller/src/opensource/nomacs.hmueller01/ImageLounge/src/DkCore/DkUtils.cpp:80:
In file included from /Library/Frameworks/Mono.framework/Headers/tiffio.h:33:
/Library/Frameworks/Mono.framework/Headers/tiff.h:77:23: error: typedef redefinition with different types ('long' vs 'int64_t' (aka 'long long'))
   77 | typedef TIFF_INT64_T  int64;
      |                       ^
/usr/local/Cellar/opencv/4.10.0_17/include/opencv4/opencv2/core/hal/interface.h:61:20: note: previous definition is here
   61 |    typedef int64_t int64;
      |                    ^
In file included from /Users/hmueller/src/opensource/nomacs.hmueller01/ImageLounge/src/DkCore/DkUtils.cpp:80:
In file included from /Library/Frameworks/Mono.framework/Headers/tiffio.h:33:
/Library/Frameworks/Mono.framework/Headers/tiff.h:78:23: error: typedef redefinition with different types ('unsigned long' vs 'uint64_t' (aka 'unsigned long long'))
   78 | typedef TIFF_UINT64_T uint64;
      |                       ^
/usr/local/Cellar/opencv/4.10.0_17/include/opencv4/opencv2/core/hal/interface.h:62:21: note: previous definition is here
   62 |    typedef uint64_t uint64;
      |                     ^
2 errors generated.
make[2]: *** [CMakeFiles/nomacsCore-3.19.1.dir/src/DkCore/DkUtils.cpp.o] Error 1
make[1]: *** [CMakeFiles/nomacsCore-3.19.1.dir/all] Error 2
make: *** [all] Error 2
```
on macOS 

I think that adding the 
```
#ifdef WITH_LIBTIFF
#ifdef Q_CC_MSVC
#include <tif_config.h>
#endif
#include <tiffio.h>
#endif
```
in DkUtils.cpp PR #1226 causes the trouble ...
Why does this work on Linux???

This is my config
```
--  nomacs - Image Lounge 3.19  <https://nomacs.org>
--  Qt version: 6.7.3
--  nomacs will be compiled with OPENCV support .................. YES
--  nomacs will be compiled with LIBRAW support .................. YES
--  nomacs will be compiled with extended TIFF support ........... YES
--  nomacs will be compiled with plugin support .................. YES
--  nomacs will be compiled with HEIF support .................... NO
--  nomacs will be compiled with Translations .................... YES
--  nomacs will be compiled with QuaZip support .................. YES
--  nomacs tests ................................................. NO
--  nomacs benchmarks ............................................ NO
```

```
> brew config
HOMEBREW_VERSION: 4.4.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3001afe88112c13b5aafdd92e097680897060881
Last commit: 3 weeks ago
Branch: stable
Core tap JSON: 21 Dec 09:46 UTC
Core cask tap JSON: 21 Dec 09:46 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0.0
HOMEBREW_EDITOR: /usr/bin/vi
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.3.6 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 16.0.0 build 1600
Git: 2.47.1 => /usr/local/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.7.1-x86_64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
```

--- a/ImageLounge/src/DkCore/DkBasicLoader.cpp
+++ b/ImageLounge/src/DkCore/DkBasicLoader.cpp
@@ -82,22 +82,18 @@
 #include <tif_config.h>
 #endif
 
-// #if defined(Q_OS_MAC) || defined(Q_OS_OPENBSD)
 //  here we clash (typedef redefinition with different types ('long' vs 'int64_t' (aka 'long long')))
 //  so we simply define our own int64 before including tiffio
 #define uint64 uint64_hack_
 #define int64 int64_hack_
-// #endif // defined(Q_OS_MAC) || defined(Q_OS_OPENBSD)
 
 #include <tiffio.h>
 #include <tiffio.hxx> // this is needed if you want to load tiffs from the buffer
 
-// #if defined(Q_OS_MAC) || defined(Q_OS_OPENBSD)
 #undef uint64
 #undef int64
-// #endif // defined(Q_OS_MAC) || defined(Q_OS_OPENBSD)
 
-#endif // #ifdef WITH_LIBTIFF
+#endif // WITH_LIBTIFF
 
 #endif // #ifdef WITH_OPENCV
 

--- a/ImageLounge/src/DkCore/DkUtils.cpp
+++ b/ImageLounge/src/DkCore/DkUtils.cpp
@@ -77,8 +77,17 @@
 #ifdef Q_CC_MSVC
 #include <tif_config.h>
 #endif
+
+//  here we clash (typedef redefinition with different types ('long' vs 'int64_t' (aka 'long long')))
+//  so we simply define our own int64 before including tiffio
+#define uint64 uint64_hack_
+#define int64 int64_hack_
+
 #include <tiffio.h>
-#endif
+
+#undef uint64
+#undef int64
+#endif // WITH_LIBTIFF
 
 #if defined(Q_OS_WIN) && !defined(SOCK_STREAM)
 #include <winsock2.h> // needed since libraw 0.16

